modules/SofaTopologyMapping/src/SofaTopologyMapping/Edge2QuadTopologicalMapping.cpp

```diff
@@ -113,10 +118,18 @@ void Edge2QuadTopologicalMapping::init()
 
             // CREATION of the points (new DOFs for the output topology) along the circles around each point of the input topology
 
+            Vec X0(1.,0.,0.);
             Vec Y0;
             Vec Z0;
-            Y0[0] = (Real) (0.0); Y0[1] = (Real) (1.0); Y0[2] = (Real) (0.0);
-            Z0[0] = (Real) (0.0); Z0[1] = (Real) (0.0); Z0[2] = (Real) (1.0);
+
+            if (ellipse){
+                Z0 = d_focalAxis.getValue();
+                Z0.normalize();
+                Y0 = cross(Z0,X0);
+            } else {
+                Y0[0] = (Real) (0.0); Y0[1] = (Real) (1.0); Y0[2] = (Real) (0.0);
+                Z0[0] = (Real) (0.0); Z0[1] = (Real) (0.0); Z0[2] = (Real) (1.0);
+            }
```

Comment:
I'm not used to this topology mapping. But normally topology mapping are only used to map 2 differents topologies together without geometry computation or creating things. 
This is more the job of an engine for me but maybe I'm too picky...